core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/WhileHandler.java

```diff
@@ -60,16 +60,20 @@ public Object doOperation(final While operation,
 
         Object input = operation.getInput();
         for (int repeatCount = 0; repeatCount < operation.getMaxRepeats(); repeatCount++) {
-            final While operationClone = operation.shallowClone();
-            if (!isSatisfied(input, operationClone, context, store)) {
+            final While clonedWhile = operation.shallowClone();
+            if (!isSatisfied(input, clonedWhile, context, store)) {
                 break;
             }
-            input = doDelegateOperation(input, operationClone.getOperation(), context, store);
+            input = doDelegateOperation(input, getOperationFromWhile(clonedWhile), context, store);
         }
 
         return input;
     }
 
+    protected Operation getOperationFromWhile(final While aWhile) {
```

Comment:
Maybe a comment to explain this is useful for overriding (in Federation)?

Reply:
I thought the rule was anything Federation should not be mentioned lower down in the Gaffer code?
Thoughts?

Reply:
I don't think we should mention Federation in the core code. It is an additional store which _shouldn't_ require changes to the core or mentions of Federation. In practice there are a few things like this. It will be easy to find out what this is for. A general JavaDoc comment would be good but a comment which just says that it's there for Federation less so.